cmd/kaspawallet/libkaspawallet/bip32/bip32_test.go

```diff
@@ -10,100 +10,192 @@ import (
 
 func TestBIP32SpecVectors(t *testing.T) {
 	type testPath struct {
-		path   string
-		extPub string
-		extPrv string
+		path               string
+		extendedPublicKey  string
+		extendedPrivateKey string
 	}
 
 	type testVector struct {
-		seed   string
-		pathes []testPath
+		seed    string
+		version [4]byte
+		paths   []testPath
 	}
 
 	// test vectors are copied from https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#Test_Vectors
 	testVectors := []testVector{
 		{
-			seed: "000102030405060708090a0b0c0d0e0f",
-			pathes: []testPath{
+			seed:    "000102030405060708090a0b0c0d0e0f",
+			version: BitcoinMainnetPrivate,
+			paths: []testPath{
```

Comment:
You have no test vectors that check public key deriviation (a.k.a capital M paths), maybe write some such as well? 
(Or alternatively, for each test-vector here, also test what happens when you replace m with M)

Reply:
TestExtendedKey_DeriveFromPath checks that.